lib/portage/package/ebuild/config.py

```diff
@@ -1650,6 +1656,23 @@ def setcpv(self, mycpv, use_cache=None, mydb=None):
 		if has_changed:
 			self.reset(keeping_pkg=1)
 
+		if explicit_iuse is None:
+			explicit_iuse = frozenset(x.lstrip("+-") for x in iuse.split())
+		if eapi_attrs.iuse_effective:
+			iuse_implicit_match = self._iuse_effective_match
+		else:
+			iuse_implicit_match = self._iuse_implicit_match
+
+		if "test" in explicit_iuse or iuse_implicit_match("test"):
+			if "test" in self.features:
+				feature_use += "test"
```

Comment:
I know this works fine but the string append without space here *looks* like a bug, so it would be nice if we could change that. Maybe something like this:
```
feature_use += (" " if feature_use else "") + "test"
```